core/server/master/src/main/java/alluxio/master/meta/MetaMasterClientServiceHandler.java

```diff
@@ -63,6 +69,33 @@ public GetServiceVersionTResponse getServiceVersion(GetServiceVersionTOptions op
     return new GetServiceVersionTResponse(Constants.META_MASTER_CLIENT_SERVICE_VERSION);
   }
 
+  @Override
+  public GetConfigReportTResponse getConfigReport(final GetConfigReportTOptions options)
+      throws TException {
+    return RpcUtils.call(LOG, (RpcUtils.RpcCallable<GetConfigReportTResponse>) () -> {
+      ConfigCheckReport report = mMetaMaster.getConfigCheckReport();
+
+      Map<alluxio.thrift.Scope, List<alluxio.thrift.InconsistentProperty>> thriftErrors
+          = new HashMap<>();
+      for (Map.Entry<PropertyKey.Scope, List<InconsistentProperty>> entry :
+          report.getConfigErrors().entrySet()) {
+        thriftErrors.put(entry.getKey().toThrift(), entry.getValue().stream()
+            .map(ThriftUtils::toThrift).collect(Collectors.toList()));
+      }
+
+      Map<alluxio.thrift.Scope, List<alluxio.thrift.InconsistentProperty>> thriftWarns
+          = new HashMap<>();
+      for (Map.Entry<PropertyKey.Scope, List<InconsistentProperty>> entry :
+          report.getConfigWarns().entrySet()) {
+        thriftWarns.put(entry.getKey().toThrift(), entry.getValue().stream()
+            .map(ThriftUtils::toThrift).collect(Collectors.toList()));
+      }
+
+      return new GetConfigReportTResponse().setErrors(thriftErrors)
```

Comment:
if we have a Thrift `ConfigCheckReport` type, we can implement this method using `ConfigCheckReport.toThrift()`.

Reply:
getConfigReport() will return a configCheckReport type.